tools/core-build-tasks/src/tasks/bundle.ts

```diff
@@ -17,20 +21,105 @@ export type BundleTaskParameters = {
+};
+
+function linkSourceMaps(
+    sourceMapDirectory: string,
+    outputDirectory: string,
+    outputFiles: OutputFile[]
+): { [key: string]: string } {
+    const generatedFiles: { [key: string]: string } = {};
+    for (const element of outputFiles) {
+        if (element.path.endsWith(MAP_EXTENSION)) {
+            const parsedPath = path.parse(element.path);
+            const sourceMapFilePath = path.join(sourceMapDirectory, parsedPath.base);
+            const sourceMapContent = JSON.parse(element.text);
+
+            // Add JS file location.
+            sourceMapContent.file = path
+                .relative(sourceMapDirectory, path.join(outputDirectory, parsedPath.name))
+                .replace(/\\/g, '/');
+            generatedFiles[sourceMapFilePath] = JSON.stringify(sourceMapContent);
+        } else {
+            // Link to the source map file.
+            const dir = path.parse(element.path).dir;
+            const targetSourceMap = path
+                .join(path.relative(dir, sourceMapDirectory), path.parse(element.path).base)
+                .replace(/\\/g, '/');
+            generatedFiles[element.path] = element.text + `\n//# sourceMappingURL=${targetSourceMap}${MAP_EXTENSION}\n`;
+        }
+    }
+
+    return generatedFiles;
+}
+
+function writeFiles(postProcessOutputFilesResult: PostProcessOutputFilesResult) {
+    fs.mkdirSync(postProcessOutputFilesResult.outputDirectory, { recursive: true });
+    if (postProcessOutputFilesResult.sourceMapDirectory !== postProcessOutputFilesResult.outputDirectory) {
+        fs.mkdirSync(postProcessOutputFilesResult.sourceMapDirectory, { recursive: true });
+    }
+
+    for (const path of Object.keys(postProcessOutputFilesResult.generatedFiles)) {
+        fs.writeFileSync(path, postProcessOutputFilesResult.generatedFiles[path]);
+    }
+}
+
+export function postProcessOutputFiles(
+    options: BundleTaskParameters,
+    buildResult: BuildResult
+): PostProcessOutputFilesResult | undefined {
+    if (!buildResult.outputFiles) {
+        return undefined;
+    }
+
+    const outputDirectory = path.parse(options.outfile).dir;
+    const sourceMapDirectory = path.resolve(options.outputSourcemapPath ?? outputDirectory);
+    const generatedFiles = linkSourceMaps(sourceMapDirectory, outputDirectory, buildResult.outputFiles);
+    return { sourceMapDirectory, outputDirectory, generatedFiles: generatedFiles };
+}
+
 export function bundleTask(options: BundleTaskParameters): ReturnType<typeof parallel> {
-    return async () => {
-        return esbuild.build({
+    return () => {
+        const buildResult: BuildResult = esbuild.buildSync({
             entryPoints: [options.entryPoint],
             bundle: true,
             format: 'esm',
             minifyWhitespace: options.minifyWhitespace,
             outfile: options.outfile,
-            sourcemap: options.sourcemap,
+            sourcemap: options.sourcemap ? 'external' : false,
```

Comment:
Does this make it so only 1 type of sourcemap is supported? Should we also support overrides for the other source map types (or sourcemap adjacent to output) but just make the default an external sourcemap.

Depends on the flexibility we want creators to have, though i'd opt for some more flexibility vs. none.

Reply:
This change is making that we support to options: no source maps or `internal` option (that is the default).
In case of generate source maps, it is using `external` option from esbuild to create the files in memory without the linking comments and post processing to append those.
Let me update the changes to support all the options.

Reply:
Still in progress

Reply:
Done, added support for all.

Reply:
And bunch of UTs